core/sql/ustat/hs_la.cpp

```diff
@@ -994,13 +999,118 @@ Lng32 HSHiveTableDef::DescribeColumnNames()
 }
 
 //=====================================================
+//
+NAString HSHbaseTableDef::getHistLoc(formatType format) const
+{
+  if ( HSGlobalsClass::isNativeHbaseCat(getCatName(format))) {
+    return HBASE_STATS_CATALOG "." HBASE_STATS_SCHEMA;
+  } else {  
+    NAString name(getCatName(format));
+    name.append(".");
+    name.append(getSchemaName(format));
+    return name;
+  }
+}
+
+static 
+Lng32 CreateExternalTable(const NAString& catName, const NAString& schName, const NAString& nativeTableName)
+{
+   HSLogMan *LM = HSLogMan::Instance();
+   if (LM->LogNeeded())
+      {
+        snprintf(LM->msg, sizeof(LM->msg), "Creating external table table for %s on demand.",
+                          nativeTableName.data());
+        LM->Log(LM->msg);
+      }
+
+   // do not have to worry about the catalog and schema for the new external table 
+   // here. These names will be determined by the processing logic. 
+   NAString ddl = "CREATE EXTERNAL TABLE ";
+   ddl.append(nativeTableName);
+   ddl.append(" FOR ");
+   ddl.append(catName);
+   ddl.append(".");
+   ddl.append(schName);
+   ddl.append(".");
+   ddl.append(nativeTableName);
+   ddl.append(" ATTRIBUTE ");
+
+   HSTranMan *TM = HSTranMan::Instance();
+   if (TM->InTransaction())
+      ddl.append(" AUDIT");
+   else
+      ddl.append("NO AUDIT");
+
```

Comment:
Good point.

On Mon, Sep 21, 2015 at 1:30 PM, Roberta Marton <notifications@github.com>
wrote:

> In core/sql/ustat/hs_la.cpp
> <https://github.com/apache/incubator-trafodion/pull/87#discussion_r40007320>
> :
>
> > +   NAString ddl = "CREATE EXTERNAL TABLE ";
> > +   ddl.append(nativeTableName);
> > +   ddl.append(" FOR ");
> > +   ddl.append(catName);
> > +   ddl.append(".");
> > +   ddl.append(schName);
> > +   ddl.append(".");
> > +   ddl.append(nativeTableName);
> > +   ddl.append(" ATTRIBUTE ");
> > +
> > +   HSTranMan *TM = HSTranMan::Instance();
> > +   if (TM->InTransaction())
> > +      ddl.append(" AUDIT");
> > +   else
> > +      ddl.append("NO AUDIT");
> > +
>
> I always assumed that external tables are non-audited. You are creating
> them either audited or non-audited. Since these tables do not contain any
> Trafodion data, is there any reason to make them audited?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/87/files#r40007320>.
>



-- 
Regards, --Qifan